### PR TITLE
Added behavior to enforce that the current date is between the given intervals at any time

### PR DESCRIPTION
**Pull request by showy**
When value < minDate or value  > maxDate the component will call onSelection passing minDate or maxDate respectively.

To activate this behavior we should pass `enforceDateIntervals=true` to the component.

The documentation for this feature has been also added to the README.

### CODE REVIEW

**Comment by duizendnegen** [comment]
What's the Pikaday behavior if minDate > value? Does it re-select?

I'll leave some in-line comments - and will update the test scripts later, removing `1.13.1`. Would you want to investigate the failing test in `ember-beta`?

File: addon/mixins/pikaday.js [hunk 1]
```diff
@@ -99,17 +104,31 @@ export default Ember.Mixin.create({
...
-      run.later(() => {
-        this.get('pikaday').setMaxDate(this.get('maxDate'));
+    const { enforceDateIntervals, pikaday, maxDate, value }  = getProperties(this, [ 'enforceDateIntervals', 'pikaday', 'maxDate', 'value' ]);
+
+    if (maxDate) {
+      run.later( () => {
+        pikaday.setMaxDate(maxDate);
+
+        // Force current value to not be greated than maxDate
+        if ( enforceDateIntervals && value > maxDate ) {
```
Comment (duizendnegen) [inline addon/mixins/pikaday.js]: I'd rather do this by default, leaving `enforceDateIntervals` out of this.

File: addon/mixins/pikaday.js [hunk 2]
```diff
@@ -99,17 +104,31 @@ export default Ember.Mixin.create({
...
+          get(this, 'onSelection')(maxDate);
```
Comment (duizendnegen) [inline addon/mixins/pikaday.js]: Shouldn't we set `value` to `maxDate` instead, and bubble the event that way?

File: tests/dummy/app/templates/application.hbs [hunk 3]
```diff
@@ -22,3 +22,10 @@ Datepicker already disabled before it's rendered:
 Inputless datepicker:
 {{pikaday-inputless onSelection=(action (mut someDate))}}
 {{someDate}}
+
+Inputless datepicker never go back:
```
Comment (duizendnegen) [inline tests/dummy/app/templates/application.hbs]: This doesn't seem like the most intuitive naming to me, what is it supposed to express?

File: addon/mixins/pikaday.js (hunk 2 above)
Comment (showy) [inline addon/mixins/pikaday.js]: Following DDUA pattern I think it's the passed function the one that should take care of setting the value.

File: tests/dummy/app/templates/application.hbs (hunk 3 above)
Comment (showy) [inline tests/dummy/app/templates/application.hbs]: Yeah, it isn't. In fact it doesn't have much of use. I will remove it.

**Comment by showy** [comment]
I'm also checking the errors for the new versions of ember.

**Comment by showy** [comment]
From what I've seem pikaday adjust it's date to mindate or maxdate if necessary, because of this I'm not doing pikaday.setDate().

File: addon/mixins/pikaday.js (hunk 1 above)
Comment (duizendnegen) [inline addon/mixins/pikaday.js]: Let's go for it anyway and release it as a minor version. Leaving the behavior off by default invites incorrect & unexpected behavior.

**Comment by duizendnegen** [comment]
Thanks for the updates - one thing remaining regarding DDAU & one regarding using this by default (I'd even go for not being able to opt-out, don't see a use case for it)

**Comment by showy** [comment]
I don't see any use case either. But I'm more worried about bugging implementations. Anyway we can check if this behavior is affecting anyone and move to a conditional activation of the behavior later on. For now I remove everything related to the `enforceDateIntervals` property.

Regarding DDUA I take your word. So now the component just calls directly pikaday.setDate().

**Comment by duizendnegen** [comment]
Thanks, appreciated!

**Pull request merged**